### ISSUE TITLE
How to add extensions?

### ISSUE THREAD

**Issue by ghost** [aptible/docker-php#9]
When using this image as a base (tag:5.6), I also included `pecl install uploadprogress` and `echo 'extension=uploadprogress.so' > /etc/php5/mods-available`, but it didn't work. Taking a look with SSH, I saw that PHP was installed to /opt, not /etc, which explains some things, but I couldn't find the mods-available folder anywhere, and php5enmod didn't seem to exist either. I'm inexperienced in every aspect of this, but I noticed that `session.upload_progress.enabled = On` is commented out in php.ini:https://github.com/aptible/docker-php/blob/master/5.6/php.ini#L1581. Is that the reason? If so, how can I fix that? Or am I just doing it wrong?

**Comment by krallin** [comment]
Hi @ralphembree,

PHP is indeed installed under `/opt` (but do note that `echo 'extension=uploadprogress.so' > /etc/php5/mods-available` likely wouldn't work even if it weren't: `mods-available` is typically a directory, so it wouldn't make sense to redirect `echo`'s output to a directory).

If you want to add a module you can append to this file instead: `/opt/php/lib/conf.d/extensions.ini`.

You can do this using the following command as part of your Docker build in an image built `FROM` our `quay.io/aptible/php` image:

```
# The $PHP_DIR environment variable references /opt/php where PHP is installed.
# Note the two >> signs to append to the file rather than overwrite.
echo 'extension=uploadprogress.so' >> "${PHP_DIR}/lib/conf.d/extensions.ini"
```

---

Regarding `php.ini`: we use the default `php.ini` from PHP. The values that are commented out are actually the default values used by PHP itself, as you can see in that file:

```
; Enable upload progress tracking in $_SESSION
; Default Value: On
; Development Value: On
; Production Value: On
; http://php.net/session.upload-progress.enabled
;session.upload_progress.enabled = On
```

In other words, this variable is actually set to on.

However, I'm not sure what you mean by asking "is that the reason"; if you'd like to clarify what you meant there, perhaps I can help.  

Cheers,

**Comment by Frederick888** [comment]
`session.upload_progress` and the `uploadprogress` extension are two different things.

To install `uploadprogress`, add the following lines to your `Dockerfile`:

```
apt-install autoconf build-essential
pecl install uploadprogress
echo "extension=uploadprogress.so" | tee -a /opt/php/lib/php.ini
```

**Comment by jacobembree** [comment]
> Is that the reason?

@krallin He meant, "Is the fact that session.upload_progress is commented out the reason that we couldn't use uploadprogress?"  Thanks, @Frederick888 for clarifying.

Thanks to both of you for helping us with this.  We used a combination of the two snippets which worked:

    apt-install autoconf build-essential
    pecl install uploadprogress
    echo 'extension=uploadprogress.so' >> "${PHP_DIR}/lib/conf.d/extensions.ini"

@ralphembree you can close this now.
